extmod/modbluetooth.c

```diff
@@ -135,83 +136,57 @@ STATIC mp_obj_t bluetooth_uuid_unary_op(mp_unary_op_t op, mp_obj_t self_in) {
     mp_obj_bluetooth_uuid_t *self = MP_OBJ_TO_PTR(self_in);
     switch (op) {
         case MP_UNARY_OP_HASH: {
-            if (self->type == MP_BLUETOOTH_UUID_TYPE_16) {
-                return mp_unary_op(MP_UNARY_OP_HASH, MP_OBJ_NEW_SMALL_INT(self->uuid._16));
-
-            } else if (self->type == MP_BLUETOOTH_UUID_TYPE_32) {
-                return mp_unary_op(MP_UNARY_OP_HASH, MP_OBJ_NEW_SMALL_INT(self->uuid._32));
-
-            } else if (self->type == MP_BLUETOOTH_UUID_TYPE_128) {
-                return MP_OBJ_NEW_SMALL_INT(qstr_compute_hash(self->uuid._128, sizeof(self->uuid._128)));
-            }
-            return MP_OBJ_NULL;
+            // Use the QSTR hash function.
+            return MP_OBJ_NEW_SMALL_INT(qstr_compute_hash(self->data, self->type));
         }
         default: return MP_OBJ_NULL; // op not supported
     }
 }
 
-
 STATIC void bluetooth_uuid_print(const mp_print_t *print, mp_obj_t self_in, mp_print_kind_t kind) {
     mp_obj_bluetooth_uuid_t *self = MP_OBJ_TO_PTR(self_in);
-
-    if (self->type == MP_BLUETOOTH_UUID_TYPE_16) {
-        mp_printf(print, "UUID16(0x%04x)", self->uuid._16);
-    } else if (self->type == MP_BLUETOOTH_UUID_TYPE_32) {
-        mp_printf(print, "UUID32(0x%08x)", self->uuid._32);
-    } else if (self->type == MP_BLUETOOTH_UUID_TYPE_128) {
-        mp_printf(print, "UUID128('");
-        for (int i = 0; i < 16; ++i) {
-            mp_printf(print, "%02x", self->uuid._128[15-i]);
-            if (i == 3 || i == 5 || i == 7 || i == 9) {
-                mp_printf(print, "-");
-            }
+    mp_printf(print, "UUID%u(%s", self->type * 8, self->type <= 4 ? "0x" : "'");
+    for (int i = 0; i < self->type; ++i) {
+        mp_printf(print, "%02x", self->data[self->type - 1 - i]);
+        if (self->type == 16 && (i == 3 || i == 5 || i == 7 || i == 9)) {
```

Comment:
This could go one line up, the `self->type==16` check removed, and 3/5/7/9 all increased by one.